app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,510 @@
+          // In the future this should be change to call a callback and update the UI without the need
+          // for a page reload
+          location.reload(true);
+        })
+        .catch(error => {
+          this.setState({ isLoading: false });
+          reportError(error);
+        });
+    });
+  };
+
+  /**
+   * Creates the action button & dropdown for each row in the table.
+   * @param {Object} data - Data about the cell this filter is called on.
+   */
+  createActionsButton = data => {
+    const rowIndex = data.rowIndex;
+    const rowData = data.rowData;
+    // Set the direction to be "up" when there are not enough rows in the table to have space for the dropdown.
+    // The table custom class handles the rest.
+    // NOTE: If this dropdown increases in height, the custom table class passed to CustomTable will need to be updated.
+    const direction = this.state.table.rowData && this.state.table.rowData.length > 4 ? null : 'up';
+    return (
+      <Dropdown drop={direction} className="close-contact-action-button">
+        <Dropdown.Toggle
+          id={`close-contact-action-button-${rowData.id}`}
+          size="sm"
+          variant="primary"
+          aria-label={`Close Contact Action Button for Close Contact ${rowData.id}`}>
+          <i className="fas fa-cogs fw" />
+        </Dropdown.Toggle>
+        <Dropdown.Menu drop={direction}>
+          <Dropdown.Item className="px-4" onClick={() => this.handleEditCC(rowIndex)}>
+            <i className="fas fa-edit fa-fw" />
+            <span className="ml-2"> Edit</span>
+          </Dropdown.Item>
+          <Dropdown.Item className="px-4" onClick={() => this.handleContactAttempt(rowIndex)}>
+            <i className="fas fa-phone fa-flip-horizontal" />
+            <span className="ml-2">Contact Attempt</span>
+          </Dropdown.Item>
+          {rowData.enrolled_id && (
+            <Dropdown.Item className="px-4" onClick={() => (location.href = `${window.BASE_PATH}/patients/${rowData.enrolled_id}`)}>
+              <i className="fas fa-search" />
+              <span className="ml-2">View Record</span>
+            </Dropdown.Item>
+          )}
+          {!rowData.enrolled_id && this.props.can_enroll_close_contacts && (
+            <Dropdown.Item className="px-4" onClick={() => (location.href = `${window.BASE_PATH}/patients/new?cc=${rowData.id}`)}>
+              <i className="fas fa-plus" />
+              <span className="ml-2"> Enroll</span>
+            </Dropdown.Item>
+          )}
+          <Dropdown.Item className="px-4" onClick={() => this.toggleCloseContactDeleteModal(rowData.id)}>
+            <i className="fas fa-trash" />
+            <span className="ml-2"> Delete</span>
+          </Dropdown.Item>
+        </Dropdown.Menu>
+      </Dropdown>
+    );
+  };
+
+  /**
+   * Gets the data for the current close contact if there is one selected/being edited.
+   */
+  getCurrentCloseContact = () => {
```

Comment:
can you move this up near the handleChange or something so it's not in between rendering methods?

Reply:
sure